### PR TITLE
Implement instance mutes

### PR DESCRIPTION
**Pull request by Tak**
Adds a menu item to the account activity, similar to the web UI, and an activity in account preferences similar to the muted users activity.

Addresses #1143

### CODE REVIEW

**Comment by connyduck** [comment]
😍

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -92,6 +93,7 @@
     <string name="action_view_favourites">Favourites</string>
     <string name="action_view_mutes">Muted users</string>
     <string name="action_view_blocks">Blocked users</string>
+    <string name="action_view_domain_mutes">Muted instances</string>
```
Comment (connyduck) [inline app/src/main/res/values/strings.xml]: the wording used in mastodon web is `Hidden domains`. I think we should follow that.

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt [hunk 2]
```diff
@@ -642,6 +655,14 @@ class AccountActivity : BottomSheetActivity(), ActionButtonActivity, HasSupportF
                 .show()
     }
 
+    private fun showMuteDomainWarningDialog(instance: String) {
+        AlertDialog.Builder(this)
+                .setMessage(getString(R.string.mute_domain_warning, instance))
+                .setPositiveButton(android.R.string.ok) { _, _ -> viewModel.muteDomain(instance) }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt]: I think the button should say "hide entire domain" like in Mastodon web.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -179,6 +182,7 @@
     <string name="dialog_unfollow_warning">Unfollow this account?</string>
     <string name="dialog_delete_toot_warning">Delete this toot?</string>
     <string name="dialog_redraft_toot_warning">Delete and re-draft this toot?</string>
+    <string name="mute_domain_warning">Are you really, really sure you want to block all of %s? In most cases, a few targeted blocks or mutes are sufficient and preferable.</string>
```
Comment (connyduck) [inline app/src/main/res/values/strings.xml]: How about

Are you sure you want to block the entire <domain>? You will not see content from that domain in any public timelines or your notifications. Your followers from that domain will be removed.

**Comment by Tak** [comment]
Ok, I think I've addressed everything except the migration to the new coding style.
(I'll still do that, but you can merge like this and I'll make a follow-up PR, if you prefer)

**Comment by connyduck** [comment]
Nice! Now, when I go to the instance mutes page while being in Airplane mode, I see the ProgressBar forever instead of the offline elephant friend.

**Comment by Tak** [comment]
Hm, I can't repro that in the emulator or on my device. Any thoughts about what's missing?

**Comment by connyduck** [comment]
Tried it again, now it works. Code also looks good. Maybe it was my mistake. 
Thx again for the awesome PR!

**Pull request merged**